ext/dom/dom.stub.php

```diff
@@ -0,0 +1,154 @@
+
+    /** @return DOMNode|false */
+    public function replaceChild(DOMNode $newChild, DOMNode $oldChild) {}
+}
+
+class DOMAttr {
+    public function __construct(string $name, ?string $value = null) {}
+    public function isId():  bool {}
+}
+
+class DOMCdataSection {
+    public function __construct(string $value) {}
+}
+
+class DOMCharacterData {
+    public function appendData(string $data):  bool {} // original $arg changed to $data
+    /** @return string|false */
+    public function substringData(int $offset, int $count) {}
+    public function insertData(int $offset, string $data):  bool {} // original $arg changed to $data
+    public function deleteData(int $offset, int $count):  bool {}
+    public function replaceData(int $offset, int $count, string $data):  bool {} // original $arg changed to $data
+}
+
+class DOMComment {
+    public function __construct(?string $value = null) {}
+}
+
+class DOMDocument {
+    public function __construct(?string $version = null, ?string $encoding = null) {}
+
+    /** @return DOMAttr|false */
+    public function createAttribute(string $name) {}
+
+    /** @return DOMAttr|false */
+    public function createAttributeNS(?string $namespaceURI = NULL, string $qualifiedName) {}
+
+    /** @return DOMCdataSection|false */
+    public function createCDATASection(string $data) {}
+
+    /** @return DOMComment|false */
+    public function createComment(string $data) {}
+
+    /** @return DOMDocumentFragment|false */
+    public function createDocumentFragment() {}
+
+    /** @return DOMElement|false */
+    public function createElement(string $tagName, ?string $value = null) {}
+
+    /** @return DOMElement|false */
+    public function createElementNS(?string $namespaceURI, string $qualifiedName, ?string $value = NULL) {}
+
+    /** @return DOMEntityReference|false */
+    public function createEntityReference(string $name) {}
+
+    /** @return DOMProcessingInstruction|false */
+    public function createProcessingInstruction(string $target, ?string $data = null) {}
+
+    /** @return DOMText|false */
+    public function createTextNode(string $data) {}
+
+    public function getElementById(string $elementId): ?DOMElement {}
+
+    public function getElementsByTagName(string $tagName): DOMNodeList {}
+
+    public function getElementsByTagNameNS(string $namespaceURI, string $localName): DOMNodeList {}
+
+    /** @return DOMNode|false */
+    public function importNode(DOMNode $importedNode, bool $deep = false) {}
+
+    /** @return DOMDocument|bool */
+    public function load(string $filename , int $options = 0) {}
+
+    /** @return DOMDocument|bool */
+    public function loadHTML(string $source, int $options = 0) {}
+
+    /** @return DOMDocument|bool */
+    public function loadHTMLFile(string $filename, int $options = 0) {}
+
+    /** @return DOMDocument|bool */
+    public function loadXML(string $source, int $options = 0) {}
+
+    public function normalizeDocument() : void {}
```

Comment:
Some functions still have a space before the return type colon that should be removed.